ddtrace/contrib/falcon/middleware.py

```diff
@@ -19,6 +22,13 @@ def __init__(self, tracer, service="falcon"):
         )
 
     def process_request(self, req, resp):
+        if self._distributed_tracing:
+            # Falcon uppercases all header names.
+            headers = dict((k.lower(), v) for k, v in req.headers.items())
```

Comment:
I use .items() here, which is a nice fast generator in Python 3 but a slower thing in Python 2.

Reply:
Yeah because it would return a list with tuples containing the copied key-value pairs of the dictionary. If we want to be smart, we can implement the same compatibility layer available in `six`: https://github.com/benjaminp/six/blob/master/six.py#L579-L608

So we'll have a kind of:
```python
# compat.py
if PY3:
    def iteritems(d, **kw):
        return iter(d.items(**kw))
else:
    def iteritems(d, **kw):
        return d.iteritems(**kw)

# this module
from .compat import iteritems
# ...
headers = dict((k.lower(), v) for k, v in iteritems(req.headers))
```

I think it should work and:
* we have a nice fast generator in Python 3
* it's the best approach in Python 2
* same call everywhere

Makes sense providing this for better performance and compatibility?

Reply:
Cool, glad there's already this wrapper so I don't need to clutter the code :)